ambry-store/src/test/java/com.github.ambry.store/IndexTest.java

```diff
@@ -1260,6 +1290,89 @@ private void testHardDeletePauseResume(boolean reloadIndex) throws InterruptedEx
     state.verifyEntriesForHardDeletes(idsDeleted);
   }
 
+  // getLogSegmentsNotInJournal helpers
+
+  /**
+   * Tests {@link PersistentIndex#getLogSegmentsNotInJournal()}
+   * @param maxLogSegmentsToBeIgnored number of log segments not to be returned via {@link PersistentIndex#getLogSegmentsNotInJournal()}
+   * @param maxEntriesInJournal max number of entries in {@link Journal}
+   * @param reloadIndex {@code true} if index needs to be reloaded and tested for {@link PersistentIndex#getLogSegmentsNotInJournal()}
+   *                    @@code false} otherwise
+   * @throws InterruptedException
+   * @throws StoreException
+   * @throws IOException
+   */
+  private void testGetLogSegmentsNotInJournal(int maxLogSegmentsToBeIgnored, int maxEntriesInJournal,
```

Comment:
I would like to keep the testGetLogSegmentsNotInJournal(int maxLogSegmentsToBeIgnored, int maxEntriesInJournal). It is very generic. Just that we are constrained by the no of log segments that could be tested as we couldn't clear the state completely and start from scratch. For eg: We can have another test case(a new one) as testGetLogSegmentsNotInJournal(5, 2 * MAX_IN_MEM_ELEMENTS); which would pass. Just that we have to work within the bounds of the max no of log segments before we could reach the max capacity.